src/component/component_system_manager.cc

```diff
@@ -244,7 +244,14 @@ tick_rotator_components(ship_space *ship) {
             pos_mat = *pos.mat;
         }
         if (*rot.rot_cur_speed) {
-            pos_mat = glm::rotate(pos_mat, (float)*rot.rot_dir * *rot.rot_cur_speed * (float)frame_info.dt, *rot.rot_axis);
+            *rot.rot_angle += *rot.rot_cur_speed;
+            // Get our angle back within range
+            while (*rot.rot_angle >= 180.0f)
+                *rot.rot_angle -= 360.0f;
+            while (*rot.rot_angle <= -180.0f)
+                *rot.rot_angle += 360.0f;
+
+            pos_mat = glm::rotate(glm::mat4(1), (float)*rot.rot_dir * *rot.rot_angle * (float)frame_info.dt, *rot.rot_axis);
```

Comment:
this doesn't seem like the right mixing in of dt

Reply:
is dt even the right thing? We could technically hit multiple times per frame, but that would presumably be to catch up on frames. So maybe it's somewhat okay. bwidawsk never finished the fixed timestep stuff, so hitting multiples times on dt is probably wrong.